vertical-pod-autoscaler/recommender/base/circularbuffer.go

```diff
@@ -0,0 +1,62 @@
+package base
+
+// A queue with a fixed maximum capacity. Elements are inserted/removed in the
+// FIFO order. Elements are removed from the buffer when it runs out of capacity
+// and a new element is inserted.
+type CircularBuffer interface {
+	// Add a value to the end of the queue. On overflow returns true and the
+	// oldest value, which is also removed from the buffer. Otherwise returns
+	// (false, _).
+	Push(value float64) (bool, float64)
+
+	// Returns the elements in the buffer, ordered by time of insertion (oldest
+	// first).
+	Contents() []float64
+
+	// Returns a pointer to the most recently added element. The pointer can be
+	// used to modify the last element. It is only valid until the next call to
+	// Push().
+	Head() *float64
```

Comment:
To allow the caller modify the last element.
This class is intended to hold a sliding window with daily memory usage peaks for a container. We will need the following 2 operations:
1. update the memory peak for today
2. progress to the next day